### ISSUE TITLE
compared with tensorforce?

### ISSUE THREAD

**Issue by jerabaul29** [rlgraph/rlgraph#5]
I am using tensorforce but I see that the development paced is reducing a bit there maybe. It fits my need well and works very fine though, so I am very happy with it and I recommend people to give it a look ;) (note: I am a user, by no ways a developer there).

Just by curiosity, what is the difference between tensorforce and rlgraph from an end-user point of view?

Of course I understand that developers of rlgraph may be a bit biaised on this question, but asking for your opinion nonetheless ;) (especially @michaelschaarschmidt who seems to work on both? ;) ).

**Comment by michaelschaarschmidt** [comment]
Hi,

thanks for asking! A main difficulty of tensorforce is debugging the end-to-end graph, and executing it in different ways (because distributed execution is tightly interwoven with model design). RLgraph has a different modularity concept which means we incorporate both TensorForce and PyTorch, and we can support different distributed backends. 

For the user, this means easier ways to scale up training, and more flexibility in combining modules , and testing them. Essentially, RLgraph will support similar algorithms initially but is thinking 'ahead' in terms large scale training and more involved algorithms. The API is otherwise very similar. 

If you are interested in the detailed design, I suggest reading the preprint describing our motivation: https://arxiv.org/abs/1810.09028

**Comment by jerabaul29** [comment]
Thank you for your answer. This is great. Very glad the API will be alike :+1: :)

**Comment by michaelschaarschmidt** [comment]
Ok, great, I will leave this open in case other users might wonder the same. Feel free to create issues for other questions.

**Comment by jerabaul29** [comment]
Good.

By the way, maybe you could put a few words about this in your readme / doc? Back a few months I had to spend 3 / 4 days to look a bit at the different RL libraries available and trying to understand their links / family relations. Would be great if this would be made clear on both tensorforce and rlgraph I think.

**Comment by michaelschaarschmidt** [comment]
Yes, will update the readme in one of the next sessions, thank you for the suggestion.
